atomic_qt_design/qml/Exchange/Exchange.qml

```diff
@@ -161,67 +166,40 @@ Item {
     }
 
     function getLastEvent(swap) {
-        if(swap.is_recent_swap && swap.events.length > 0) {
+        if(swap.is_swap && swap.events.length > 0) {
             return swap.events[swap.events.length-1]
         }
 
         return { state: '', data: { error: '' } }
     }
 
-    function getStatus(swap) {
-        if(!swap.is_recent_swap && !swap.am_i_maker) return status_swap_matching
-        if(!swap.is_recent_swap) return status_swap_not_swap
-
-        const last_state = swap.events[swap.events.length-1].state
-
-        if(last_state === "Started") return status_swap_matched
-        if(last_state === "Finished") return getSwapError(swap).state === '' ? status_swap_successful : status_swap_failed
-
-        return status_swap_ongoing
-    }
-
-    function getStatusColor(swap) {
-        const status = getStatus(swap)
-        return status === status_swap_matching ? Style.colorYellow :
-               status === status_swap_matched ? Style.colorOrange :
-               status === status_swap_ongoing ? Style.colorOrange :
-               status === status_swap_successful ? Style.colorGreen : Style.colorRed
-    }
-
-    function getStatusText(swap) {
-        const status = getStatus(swap)
-        return status === status_swap_matching ? qsTr("Order Matching") :
-                status === status_swap_matched ? qsTr("Order Matched") :
-                status === status_swap_ongoing ? qsTr("Swap Ongoing") :
-                status === status_swap_successful ? qsTr("Swap Successful") :
-                                                        qsTr("Swap Failed")
+    function getStatusColor(status) {
+        return status === "matching" ? Style.colorYellow :
+               status === "matched" ? Style.colorOrange :
+               status === "ongoing" ? Style.colorOrange :
+               status === "refunding" ? Style.colorOrange :
```

Comment:
why not "group" these with an OR?

Reply:
It's cleaner to see and modify which status has what. It's not a performance sensitive code part so I prefer clarity. This way they are same as other functions like `getStatusText` and `getStatusStep`. Those won't have that `OR` for those states. I can still do that if you want, using a switch is also possible but this code will be shorter.

Reply:
Turned them to switch cases.